tests/test_election.py

```diff
@@ -0,0 +1,260 @@
+"""
+Unit Tests for Election Models
+"""
+
+import pytest
+import vcr
+from blaseball_mike.models import ElectionResult, Election, DecreeResult, BlessingResult, TidingResult, Team
+from .helpers import TestBase, CASSETTE_DIR
+
+
+class TestElection(TestBase):
+    def test_base_compliance(self, election):
+        self.base_test(election)
+
+    def test_current_election(self, election):
+        """Test that the currently running Election does not error"""
+        assert isinstance(election, Election)
+        assert isinstance(election.decrees, list)
+        assert isinstance(election.blessings, list)
+
+    @pytest.fixture(scope="module")
+    @vcr.use_cassette(f'{CASSETTE_DIR}/Fixture.election_current.yaml')
+    def election_current(self):
+        """Current Election data"""
+        return Election.load()
+
+    @pytest.fixture(scope="module", params=['election_current'])
+    def election(self, request):
+        """Parameterized fixture of various elections"""
+        return request.getfixturevalue(request.param)
+
+
+class TestElectionResults(TestBase):
+    def test_base_compliance(self, election_result):
+        self.base_test(election_result)
+
+    @pytest.mark.vcr
+    def test_election_results(self, election_result):
+        """Test that previous season Election results do not error"""
+        assert isinstance(election_result, ElectionResult)
+
+        assert isinstance(election_result.id, str)
+        assert isinstance(election_result.name, str)
+        assert isinstance(election_result.total_bonus_votes, int)
+        assert isinstance(election_result.total_decree_votes, int)
+        assert isinstance(election_result.vote_count, int)
+
+        assert isinstance(election_result.event_results, list)
+        assert isinstance(election_result.blessing_results, list)
+        assert isinstance(election_result.decree_results, list)
+
+        for blessing in election_result.bonus_results:
+            assert isinstance(blessing, BlessingResult)
+
+        for decree in election_result.decree_results:
+            assert isinstance(decree, DecreeResult)
+
+        for tiding in election_result.tiding_results:
+            assert isinstance(tiding, TidingResult)
+
+    @pytest.mark.vcr
+    def test_load_by_season(self):
+        result = ElectionResult.load_by_season(1)
+        assert isinstance(result, ElectionResult)
+        # TODO: oops whoops uh oh
```

Comment:
uh?

Reply:
Yet another case of us forgetting to 1-index a season. But fixing it'd be a breaking change and I'm not sure it's worth doing another Major release since we just did one last week.

Reply:
i'd consider fixing an off by one to be a bug fix tbh, so go ahead and fix it

Reply:
Wasnt sure on this one, but it's fixed. Realistically it's impossible for someone to get the `ElectionResults` without already knowing what season they're looking up anyway.